iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/HelmPreprocessor.java

```diff
@@ -19,14 +19,59 @@
  */
 package org.sonar.iac.kubernetes.plugin;
 
+import com.google.protobuf.InvalidProtocolBufferException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.iac.kubernetes.jna.Loader;
+import org.sonar.iac.kubernetes.jna.library.IacHelmLibrary;
+import org.sonarsource.iac.helm.TemplateEvaluationResult;
+
+import javax.annotation.Nullable;
 import java.util.regex.Pattern;
 
 public final class HelmPreprocessor {
 
+  private static final Logger LOG = LoggerFactory.getLogger(HelmPreprocessor.class);
   private static final String NEW_LINE = "\\n\\r\\u2028\\u2029";
   private static final Pattern LINE_PATTERN = Pattern.compile("(?<lineContent>[^" + NEW_LINE + "]*+)(?<newLine>\\r\\n|[" + NEW_LINE + "])");
+  @Nullable
+  private final IacHelmLibrary iacHelmLibrary;
+
+  public HelmPreprocessor() {
+    IacHelmLibrary library;
+    try {
+      library = Loader.load("/sonar-helm-for-iac", IacHelmLibrary.class);
+    } catch (RuntimeException e) {
+      LOG.debug("Native library not loaded, Helm integration will be disabled", e);
+      library = null;
+    }
+    this.iacHelmLibrary = library;
+  }
 
-  private HelmPreprocessor() {
+  public String evaluateTemplate(String path, String content, String valuesFileContent) {
+    if (iacHelmLibrary == null || valuesFileContent.isBlank()) {
+      LOG.debug("Template cannot be evaluated, skipping processing of Helm file {}", path);
+      return "{}";
+    }
+    var rawEvaluationResult = iacHelmLibrary.evaluateTemplate(path, content, valuesFileContent).getByteArray();
```

Comment:
It will be good when `evaluateTemplate` returns `TemplateEvaluationResult` and here will be only the validation of it.